app/models/discourse_post_event/event.rb

```diff
@@ -365,5 +367,21 @@ def calculate_next_date
 
       { starts_at: next_starts_at, ends_at: next_ends_at }
     end
+
+    private
+
+    def enforce_recurrence
+      valid_recurrences = %w[
+        every_month
+        every_week
+        every_two_weeks
+        every_day
+        every_weekday
+      ]
+
+      if self.recurrence.present? && !valid_recurrences.include?(self.recurrence)
+        self.recurrence = 'every_week'
+      end
```

Comment:
> Due to the fact we do this in :post_process_cooked

`post_process_cooked` -> `update_from_raw(post)` -> ...

Do you mean https://github.com/discourse/discourse-calendar/blob/b7e0441e77b64d1230faa9a72deda0c138d80ff3/app/models/discourse_post_event/event.rb#L333-L367?

Does this mean we have duplication?